File: backend/IDE.Common/ModelsDTO/DTO/Common/NotificationDTO.cs
using IDE.Common.Enums;
using System;

namespace IDE.Common.ModelsDTO.DTO.Common
{
    public class NotificationDTO
    {
        public int Id { get; set; }
        public NotificationStatus Status { get; set; }
        public NotificationType Type { get; set; }
        public DateTime DateTime { get; set; }
        public string Message { get; set; }
        public bool IsRead { get; set; }
        public int? ProjectId { get; set; }
        public string Metadata { get; set; }
    }
}

File: frontend/src/app/models/DTO/Common/notificationDTO.ts
/* Auto Generated */

import { NotificationStatus } from "./../../Enums/notificationStatus"
import { NotificationType } from "./../../Enums/notificationType"

export interface NotificationDTO {
    id: number;
    status: NotificationStatus;
    type: NotificationType;
    dateTime: Date;
    message: string;
    isRead: boolean;
    projectId?: number;
    metadata: string;
}
